examples/images_contours_and_fields/image_antialiasing.py

```diff
@@ -72,6 +74,73 @@
     ax.set_title(f"interpolation='{interp}'")
 plt.show()
 
+###############################################################################
+# Data antialiasing and RGBA antialiasing
+# ----------------------------------------
+#
+# The examples above all used grayscale.  When colormapping is added there
+# is the complication that downsampling and the antialiasing filters are
+# applied to the data in Matplotlib, before the data is mapped to
+# colors.  So in the following note how the corners fade to white, the middle
+# of the colormap, because the data is being smoothed and the high and low
+# values are averaging out to the middle of the colormap.
+
+f0 = 10
+k = 150
+a = np.sin(np.pi * 2 * (f0 * R + k * R**2 / 2))
+
+fig, axs = plt.subplots(1, 2, figsize=(7, 4), sharex=True, sharey=True,
+                        constrained_layout=True)
+for ax, interp in zip(axs, ['nearest', 'antialiased']):
+    pc = ax.imshow(a, interpolation=interp, cmap='RdBu_r', vmin=-1, vmax=1)
+    ax.set_title(f"'{interp}'", fontsize='small')
+fig.colorbar(pc, ax=axs)
+plt.show()
+
+###############################################################################
+# Sometimes, however, we want the antialiasing to occur in RGBA space.  In
+# this case purple is actually the perceptual mixture of red and blue.
+# Matplotlib doesn't allow this to be directly achieved, but we can pass
+# RGBA data to `~.Axes.imshow`, and then the antialiasing filter will be
+# applied to the RGBA data:
+
+fig, axs = plt.subplots(1, 2, figsize=(7, 4), sharex=True, sharey=True,
+                        constrained_layout=True)
+norm = mcolors.Normalize(vmin=-1, vmax=1)
+cmap = cm.RdBu_r
+a_rgba = cmap(norm(a))
+for ax, interp in zip(axs, ['nearest', 'antialiased']):
+    pc = ax.imshow(a_rgba, interpolation=interp)
+    ax.set_title(f"'{interp}'", fontsize='small')
+plt.show()
+
+###############################################################################
+# A concrete example of where antialiasing in data space may not be desirable
+# is given here, where the anti-aliasing returns white pixels in data space,
+# and (imperceptible) purple pixels in RGBA space:
```

Comment:
```suggestion
# is given here: All data points are either -1 or +1, corresponding to blue and red. The smoothing in data space can create any value between -1 and +1, and thus result in values that never occur in the original data (e.g. 0-valued pixels in white). Alternatively, when smoothing in RGBA space, we can get purple pixels as a mixture of blue and red. The disadvantage here is that these colors are not even in the colormap, and thus strictly have no meaning. However, visually they have a more intuitive "mixture of blue and red" appearance.
```

Reply:
OK, modified this based on this suggestion....